kyo-prelude/shared/src/main/scala/kyo2/kernel/Effect.scala

```diff
@@ -60,11 +60,11 @@ object Effect:
         )(
             inline handle: Safepoint ?=> [C] => (I[C], O[C] => A < (E & S & S2)) => A < (E & S & S2),
             inline done: A => B < S3 = (v: A) => v,
-            accept: [C] => (I[C], Tag[Any]) => Boolean = [C] => (v: I[C], tag2: Tag[Any]) => tag =:= tag2
+            inline accept: [C] => I[C] => Boolean = [C] => (v: I[C]) => true
```

Comment:
There was a bug in `Result.run` because the provided `accept` function was ignoring the `Tag`, which was matching inputs of other effects. I've changed `Effect.handle` to always check the tag and allow `accept` to check only the input.